BREAKING_CHANGES.md

```diff
@@ -1,38 +1,303 @@
+The `assetsDir` field was [deprecated in Stencil v2.0.0](#componentassetsdir), but some backwards compatibility was retained with a warning message.
+It has been fully removed in Stencil v3.0.0 in favor of `assetsDirs`.
+To migrate from existing usages of `assetsDir`, update the property name and wrap its value in an array:
+```diff
+@Component({
+  tag: 'my-component',
+- assetsDir: 'assets',
++ assetsDirs: ['assets'],
+})
+```
+For more information on the `assetsDirs` field, please see the [Stencil Documentation on `assetsDirs`](https://stenciljs.com/docs/assets#assetsdirs)
+
+#### Drop Node 12 Support
+Stencil no longer supports Node 12.
+Please upgrade local development machines, continuous integration pipelines, etc. to use Node v14 or higher.
+
+#### Strongly Typed Inputs
+`onInput` and `onInputCapture` events have had their interface's updated to accept an argument of `InputEvent` over `Event`:
+```diff
+- onInput?: (event: Event) => void;
++ onInput?: (event: InputEvent) => void;
+- onInputCapture?: (event: Event) => void;
++ onInputCapture?: (event: InputEvent) => void;
+```
+`event` arguments to either callback should be updated to take this narrower typing into account
+
+#### Narrowed Typing for `autocapitalize` Attribute
+The [`autocaptialize` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autocapitalize) has been narrowed from type `any` to type `string`.
+This changes brings Stencil into closer alignment with TypeScript's typings for the attribute.
+No explicit changes are needed, unless a project was passing non-strings to the attribute.
+
+### Custom Types for Props and Events are now Exported from `components.d.ts`
+
+Custom types for props and custom events are now re-exported from a project's `components.d.ts` file.
+
+For the following Stencil component
+```tsx
+import { Component, Event, EventEmitter, Prop, h } from '@stencil/core';
+
+export type NameType = string;
+export type Todo = Event;
+
+@Component({
+  tag: 'my-component',
+  styleUrl: 'my-component.css',
+  shadow: true,
+})
+export class MyComponent {
+  @Prop() first: NameType;
+
+  @Event() todoCompleted: EventEmitter<Todo>
+
+  render() {
+    return <div>Hello, World! I'm {this.first}</div>;
+  }
+}
+```
+
+The following data will now be included automatically in `components.d.ts`:
+```diff
+ import { HTMLStencilElement, JSXBase } from "@stencil/core/internal";
++import { NameType } from "./components/my-component/my-component";
++export { NameType } from "./components/my-component/my-component";
+ export namespace Components {
+     interface MyComponent {
+-        "first": string;
++        "first": NameType;
+     }
+ }
++export interface MyComponentCustomEvent<T> extends CustomEvent<T> {
++    detail: T;
++    target: HTMLMyComponentElement;
++}
+ declare global {
+     interface HTMLMyComponentElement extends Components.MyComponent, HTMLStencilElement {
+}
+```
```

Comment:
The types were already getting imported, the only change is that they get re-exported now.

```suggestion
```diff
 import { HTMLStencilElement, JSXBase } from "@stencil/core/internal";
 import { NameType } from "./components/my-component/my-component";
+export { NameType } from "./components/my-component/my-component";
 export namespace Components {
     interface MyComponent {
        "first": NameType;
     }
 }
 export interface MyComponentCustomEvent<T> extends CustomEvent<T> {
     detail: T;
     target: HTMLMyComponentElement;
 }
 declare global {
     interface HTMLMyComponentElement extends Components.MyComponent, HTMLStencilElement {
}
```
```

Reply:
@tanner-reits can you do me a favor and transfer this over to https://github.com/ionic-team/stencil-site/pull/922?

This PR is still a draft/isn't ready to be reviewed yet - I wanted to use the stencil-site version as the source of truth for review comments (then once that PR is good, I'll transfer things over to this one)